pkg/kube/pod_writer.go

```diff
@@ -53,7 +53,7 @@ func NewPodWriter(cli kubernetes.Interface, path string, content io.Reader) PodW
 
 // Write will create a new file(if not present) and write the provided content to the file
 func (p *podWriter) Write(ctx context.Context, namespace, podName, containerName string) error {
-	cmd := []string{"sh", "-c", "cat - > " + p.path}
+	cmd := []string{"sh", "-c", "cat - > " + p.path + " && :"}
```

Comment:
can this result into a problem, if someone wrote something and immediately after it returns they tried to check the content?

Reply:
That's the issue currently. In `pod_writer_test.go` we have a flake because we execute `Write` and then check the content right away, which gets us an empty file. Even worse that adding sleep in the test didn't fix it sometimes, suggesting that exec finished before it could process the input data (that's how I understood it).
My understanding is with the fix `cat` waits for the input and `Write` only returns when the data is written to the file.